src/app/components/event/list/EventList.vue

```diff
@@ -28,31 +68,91 @@ const props = withDefaults(defineProps<Props>(), {
   accountId: undefined,
 })
 
-const { t } = useI18n()
-
-// refs
-const after = ref<string>()
-
-// api data
 const eventsQuery = await zalgo(
   useAllEventsQuery({
     after,
     createdBy: props.accountId,
     first: ITEMS_PER_PAGE,
   }),
 )
-const api = getApiData([eventsQuery])
-const events = computed(
-  () =>
-    eventsQuery.data.value?.allEvents?.nodes
-      .map((x) => getEventItem(x))
-      .filter(isNeitherNullNorUndefined) || [],
+
+const debouncedSearchQuery = useDebounce(searchQuery, 300)
+
+const searchLanguage = computed(() =>
+  locale.value === 'de' ? Language.De : Language.En,
+)
+
+const computedSearchVars = computed(() => {
+  const trimmed = debouncedSearchQuery.value.trim()
+  if (trimmed.length < 2) return null
+
+  let searchTerm = trimmed
+  const wordsArray: string[] = trimmed.split(' ').filter(Boolean)
+  if (wordsArray.length === 1 && wordsArray[0] && wordsArray[0].length < 5) {
+    searchTerm = `${wordsArray[0]}* ${wordsArray[0]}`
+  }
+  return { query: searchTerm, language: searchLanguage.value }
+})
+
+const searchVars = ref<{ query: string; language: Language } | null>(null)
+watch(
+  computedSearchVars,
+  (newVars) => {
+    searchVars.value = newVars
+  },
+  { immediate: true },
 )
+
+const searchResultsQuery = useEventSearchQuery(
+  () => ({
+    query: searchVars.value?.query || '',
+    language: searchVars.value?.language || searchLanguage.value,
```

Comment:
Right now all events don't have a language assigned to them and we'll keep it this way for now. This means the search tokens are generated by the backend without considering any language. Setting the language as search parameter currently doesn't work well with this. Let's leave out the language specification here.